### PR TITLE
fuzz: adds validation target for maybe invalid module compilation

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

File: internal/wasm/func_validation.go [hunk 1]
```diff
@@ -1464,6 +1464,9 @@ func (m *Module) validateFunctionWithMaxStackValues(
 				valueTypeStack.push(p)
 			}
 		} else if op == OpcodeEnd {
+			if len(controlBlockStack) == 0 {
```
Comment (mathetake) [inline internal/wasm/func_validation.go:1467]: found a bug actually :D

File: internal/integration_test/fuzz/wazerolib/nodiff.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
+package main
```
Comment (mathetake) [inline internal/integration_test/fuzz/wazerolib/nodiff.go:1]: this is just moved from lib.go

File: internal/wasm/func_validation.go (hunk 1 above)
Comment (mathetake) [reply internal/wasm/func_validation.go:1467]: oh maybe I need to add regression test case

File: internal/integration_test/fuzz/fuzz/fuzz_targets/validation.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+fuzz_target!(|data: &[u8]| {
+    drop(run(data));
+});
+
+fn run(data: &[u8]) -> Result<()> {
+    // Create the random source.
+    let mut u = Unstructured::new(data);
+
+    // Generate the random module via wasm-smith, but MaybeInvalidModule.
+    let module: wasm_smith::MaybeInvalidModule = u.arbitrary()?;
```
Comment (codefromthecrypt) [inline internal/integration_test/fuzz/fuzz/fuzz_targets/validation.rs:15]: so this isn't sure if it will make an invalid one or not.. interesting :D

**Review by codefromthecrypt — APPROVED** [review]
agree failing earlier is better especially as wasm gives us a lot of validation opportunity

File: internal/wasm/func_validation.go (hunk 1 above)
Comment (codefromthecrypt) [reply internal/wasm/func_validation.go:1467]: yeah thanks I should have nagged you, but you self-nagged. excellent

File: internal/integration_test/fuzz/wazerolib/validate_test.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	"testing"
+)
+
+// TestReRunFailedValidateCase re-runs the failed case specified by WASM_BINARY_NAME in testdata directory.
+func TestReRunFailedValidateCase(t *testing.T) {
+	binaryPath := os.Getenv("WASM_BINARY_PATH")
+
+	wasmBin, err := os.ReadFile(binaryPath)
+	if err != nil {
+		t.Fatal(err)
```
Comment (codefromthecrypt) [inline internal/integration_test/fuzz/wazerolib/validate_test.go:14]: should this skip or fail?

File: internal/integration_test/fuzz/fuzz/fuzz_targets/validation.rs (hunk 3 above)
Comment (mathetake) [reply internal/integration_test/fuzz/fuzz/fuzz_targets/validation.rs:15]: yeah not sure until we pass it to wazero :D

**Pull request merged**